File: clientPython/zeroc-ice-3.7.2/csharp/test/Ice/faultTolerance/TestI.cs
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

using System.Diagnostics;
using System.Threading;
using Test;

public sealed class TestI : TestIntfDisp_
{
    public TestI()
    {
        lock(this)
        {
            _p = Process.GetCurrentProcess();
            _pid = _p.Id;
        }
    }

    private void commitSuicide()
    {
        _p.Kill();
        Thread.Sleep(5000); // Give other threads time to die.
    }

    public override void abort(Ice.Current current)
    {
        commitSuicide();
    }

    public override void idempotentAbort(Ice.Current current)
    {
        commitSuicide();
    }

    public override int pid(Ice.Current current)
    {
        lock(this)
        {
            return _pid;
        }
    }

    public override void shutdown(Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }

    private Process _p;
    private int _pid;
}

File: clientPython/zeroc-ice-3.7.2/java/test/src/main/java/test/Ice/faultTolerance/TestI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package test.Ice.faultTolerance;

import test.Ice.faultTolerance.Test.TestIntf;

public final class TestI implements TestIntf
{
    public TestI(int port)
    {
        _pseudoPid = port; // We use the port number instead of the process ID in Java.
    }

    @Override
    public void shutdown(com.zeroc.Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }

    @Override
    public void abort(com.zeroc.Ice.Current current)
    {
        Runtime.getRuntime().halt(0);
    }

    @Override
    public void idempotentAbort(com.zeroc.Ice.Current current)
    {
        Runtime.getRuntime().halt(0);
    }

    @Override
    public int pid(com.zeroc.Ice.Current current)
    {
        return _pseudoPid;
    }

    private int _pseudoPid;
}
